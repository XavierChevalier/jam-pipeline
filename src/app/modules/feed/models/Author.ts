import { InferType, object, string } from 'yup'

export const Author = object({
  name: string().required(),
  avatar: string().required().url(),
})

export type Author = InferType<typeof Author>
