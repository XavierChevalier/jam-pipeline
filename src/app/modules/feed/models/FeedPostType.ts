export enum AvailableFeedPostType {
  userShareSong = 'userShareSong',
  userShareAlbum = 'userShareAlbum',
  userSharePlaylist = 'userSharePlaylist',
  artistReleaseSong = 'artistReleaseSong',
  artistReleaseAlbum = 'artistReleaseAlbum',
}
