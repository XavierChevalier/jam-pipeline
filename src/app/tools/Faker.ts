export const generateImageUrl = (size = 50) => `https://picsum.photos/${size}`
