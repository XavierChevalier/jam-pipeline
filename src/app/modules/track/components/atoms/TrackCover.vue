<script lang="ts">
  import { defineComponent } from 'vue'
  import { SizeMixin } from '@/app/mixins/SizeMixin'

  export default defineComponent({
    mixins: [
      SizeMixin({
        sm: 'w-9 h-9',
        base: 'w-16 h-16',
      }),
    ],
  })
</script>

<script setup lang="ts">
  import { TrackOfTrackCover } from '@/app/modules/track/models/TrackCover'
  import { computed, PropType } from 'vue'
  import { isPropertyValid } from '@/app/tools/component-properties/PropertyValidator'

  const props = defineProps({
    track: {
      type: Object as PropType<TrackOfTrackCover>,
      required: true,
      validator: isPropertyValid(TrackOfTrackCover),
    },
  })

  const alt = computed(() => {
    const bands = props.track.bands.join(', ')

    return `The track cover of ${props.track.title} â€¢ ${bands}`
  })
</script>

<template>
  <img class="rounded" :class="sizeClass" :src="track.cover" :alt="alt" />
</template>
