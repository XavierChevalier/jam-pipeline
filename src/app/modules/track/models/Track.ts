import { TrackCover } from '@/app/modules/track/models/TrackCover'
import { array, InferType, number, object, string } from 'yup'

export const TrackTitle = string().required()
export const TrackBands = array(string().required()).required().min(1)

export const Track = object({
  cover: TrackCover,
  title: TrackTitle,
  bands: TrackBands,
  durationMs: number().positive().required(),
})
export type Track = InferType<typeof Track>
