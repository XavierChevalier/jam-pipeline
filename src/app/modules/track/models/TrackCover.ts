import { TrackBands, TrackTitle } from '@/app/modules/track/models/Track'
import { InferType, object, string } from 'yup'

export const TrackCover = string().url().required()
export type TrackCover = InferType<typeof TrackCover>

export const TrackOfTrackCover = object({
  cover: TrackCover,
  title: TrackTitle,
  bands: TrackBands,
})
export type TrackOfTrackCover = InferType<typeof TrackOfTrackCover>
