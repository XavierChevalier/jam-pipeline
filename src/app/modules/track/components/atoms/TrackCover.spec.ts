import * as Stories from '@/app/modules/track/components/atoms/TrackCover.stories'
import { TrackOfTrackCover } from '@/app/modules/track/models/TrackCover'
import { omit } from 'lodash-es'
import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'
import { generateImageUrl } from '@/app/tools/Faker'

describe('TrackCover', () => {
  StorybookTestsGenerator.fromStoriesExports(Stories)
    .snapshotEachStories()
    .property<TrackOfTrackCover>('track', (property) => {
      const track = {
        cover: generateImageUrl(),
        title: 'G.O.A.T.',
        bands: ['Polyphia'],
      }
      property
        .itShouldBeDefined()
        .itShouldBeRequired()
        .itShouldHaveAValidatorFunction()
        .itShouldSuccessWith(track)
        .itShouldFailWith('missing cover', omit(track, 'cover'))
        .itShouldFailWith('missing title', omit(track, 'title'))
        .itShouldFailWith('missing bands', omit(track, 'bands'))
    })
    .property<string>('size', (property) => {
      property
        .itShouldBeDefined()
        .itShouldBeTypeOf(String)
        .itShouldHaveADefaultValue()
        .itShouldHaveAValidatorFunction()
        .itShouldSuccessWith('sm')
        .itShouldSuccessWith('base')
        .itShouldFailWith('unknown size type', 'unknown')
    })
})
