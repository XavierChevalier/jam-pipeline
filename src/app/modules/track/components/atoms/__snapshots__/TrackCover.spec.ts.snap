// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackCover stories snapshots [snapshot] should render the "Default" component story correctly 1`] = `
<img
  alt="The track cover of G.O.A.T. • Polyphia"
  class="rounded w-16 h-16"
  src="https://picsum.photos/50"
/>
`;

exports[`TrackCover stories snapshots [snapshot] should render the "SmallSize" component story correctly 1`] = `
<img
  alt="The track cover of G.O.A.T. • Polyphia"
  class="rounded w-9 h-9"
  src="https://picsum.photos/50"
/>
`;
